xbox/xbox.c

```diff
@@ -0,0 +1,1002 @@
+  p = xgu_set_blend_func_sfactor(p, gl_to_xgu_blend_factor(sfactor));
+  p = xgu_set_blend_func_dfactor(p, gl_to_xgu_blend_factor(dfactor));
+  pb_end(p);
+}
+
+GL_API void GL_APIENTRY glClipPlanef (GLenum p, const GLfloat *eqn) {
+  //FIXME: Use a free texture stage and set it to cull mode
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glColor4ub (GLubyte red, GLubyte green, GLubyte blue, GLubyte alpha) {
+  uint32_t* p = pb_begin();
+  p = xgux_set_color4ub(p, red, green, blue, alpha);
+  pb_end(p);
+}
+
+GL_API void GL_APIENTRY glColor4f (GLfloat red, GLfloat green, GLfloat blue, GLfloat alpha) {
+  uint32_t* p = pb_begin();
+  p = xgux_set_color4f(p, red, green, blue, alpha);
+  pb_end(p);
+}
+
+GL_API void GL_APIENTRY glColorMask (GLboolean red, GLboolean green, GLboolean blue, GLboolean alpha) {
+
+  XguColorMask mask = 0;
+  if (red   != GL_FALSE) { mask |= XGU_RED;   }
+  if (green != GL_FALSE) { mask |= XGU_GREEN; }
+  if (blue  != GL_FALSE) { mask |= XGU_BLUE;  }
+  if (alpha != GL_FALSE) { mask |= XGU_ALPHA; }
+
+  uint32_t* p = pb_begin();
+  p = xgu_set_color_mask(p, mask);
+  pb_end(p);
+}
+
+GL_API void GL_APIENTRY glDepthFunc (GLenum func) {
+  //FIXME: Missing from XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glDepthMask (GLboolean flag) {
+  //FIXME: Missing from XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glEnable (GLenum cap) {
+  set_enabled(cap, true);
+}
+
+GL_API void GL_APIENTRY glDisable (GLenum cap) {
+  set_enabled(cap, false);
+}
+
+GL_API void GL_APIENTRY glEnableClientState (GLenum array) {
+  set_client_state_enabled(array, true);
+}
+
+GL_API void GL_APIENTRY glDisableClientState (GLenum array) {
+  set_client_state_enabled(array, false);
+}
+
+GL_API void GL_APIENTRY glCullFace (GLenum mode) {
+  uint32_t* p = pb_begin();
+  p = xgu_set_cull_face(p, gl_to_xgu_cull_face(mode));
+  pb_end(p);
+}
+
+GL_API void GL_APIENTRY glFrontFace (GLenum mode) {
+  uint32_t* p = pb_begin();
+  p = xgu_set_front_face(p, gl_to_xgu_front_face(mode));
+  pb_end(p);
+}
+
+
+// Stencil actions
+GL_API void GL_APIENTRY glStencilFunc (GLenum func, GLint ref, GLuint mask) {
+  //FIXME: Missing from XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glStencilOp (GLenum fail, GLenum zfail, GLenum zpass) {
+  uint32_t* p = pb_begin();
+  p = xgu_set_stencil_op_fail(p, gl_to_xgu_stencil_op(fail));
+  p = xgu_set_stencil_op_zfail(p, gl_to_xgu_stencil_op(zfail));
+  p = xgu_set_stencil_op_zpass(p, gl_to_xgu_stencil_op(zpass));
+  pb_end(p);
+}
+
+
+// Misc.
+GL_API void GL_APIENTRY glPointParameterf (GLenum pname, GLfloat param) {
+  //FIXME: Bad in XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glPointParameterfv (GLenum pname, const GLfloat *params) {
+  //FIXME: Bad in XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glPointSize (GLfloat size) {
+  //FIXME: Bad in XGU
+  assert(false);
+}
+
+GL_API void GL_APIENTRY glPolygonOffset (GLfloat factor, GLfloat units) {
+  //FIXME: Missing from XGU
```

Comment:
Not really important (and potentially tricky); but still waiting for this CC @dracc